Framework/Algorithms/test/SofQWTest.h

```diff
@@ -117,6 +133,19 @@ class SofQWTest : public CxxTest::TestSuite {
     const auto child = lastAlg->getChildAlgorithmHistory(0);
     return (child->name() == name);
   }
+
+  bool isNanInHistogram(const Mantid::API::MatrixWorkspace &result,
+                        const size_t index) {
+    bool nanFound = false;
+    for (size_t i = 0; i < result.blocksize(); i++) {
+      if (isnan(result.y(index)[i]) || isnan(result.e(index)[i])) {
+        nanFound = true;
+        break; // NaN found in histogram, no need to keep searching
+      }
+    }
```

Comment:
I think we are trying to avoid repeatedly obtaining a reference to a vector in a tight loop. The loop can be replaced with `std::find`. on each vector.